includes/tracks/events/class-wc-admin-setup-wizard-tracking.php

```diff
@@ -19,9 +19,47 @@ public static function init() {
 			return;
 		}
 
+		add_action( 'wc_setup_footer', array( __CLASS__, 'add_footer_scripts' ) );
+		add_action( 'admin_init', array( __CLASS__, 'track_marketing_signup' ), 1 );
 		self::add_step_save_events();
 	}
 
+	/**
+	 * Get the name of the current step.
+	 *
+	 * @return string
+	 */
+	public static function get_current_step() {
+		return isset( $_GET['step'] ) ? sanitize_key( $_GET['step'] ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
+	}
+
+	/**
+	 * Add footer scripts to OBW since it does not contain hooks for
+	 * wp_footer to allow the default methods of enqueuing scripts.
+	 */
+	public static function add_footer_scripts() {
```

Comment:
Instead of echoing the script, this function can be:

```php
public static function print_scripts() {
    wp_print_scripts();
    wc_print_js();
}
```

Reply:
The problem with using `wp_print_scripts()` is that it adds a bunch of other unwanted scripts to this page.  Some harmless, but some that introduce odd behavior (also depending on the plugins/theme you have installed).

Added a whitelist function in https://github.com/woocommerce/woocommerce/pull/22903/commits/d8ec3a1b1bacd046d729c228e5b6f6fa3fc9ba35 so we can do this more the WP-way.  Let me know if that looks a bit better.

Reply:
This looks way better. 

FYI: I'm changing the use of `wc_enqueue_js` in https://github.com/woocommerce/woocommerce/pull/22920. I _think_ this means we won't need `wc_print_js`, but I'm not sure. We may need to address this depending on which PR gets merged first.

Reply:
Merged in the tracking script scope PR here - https://github.com/woocommerce/woocommerce/pull/22914